src/components/TransactionPreview/ActionPreview.tsx

```diff
@@ -462,7 +462,15 @@ const ActionPreview = ({
 
   const cost = useMemo(() => {
     if (isEstimating) return "Estimating...";
-    if (!estimated || !estimated?.gasCost) return estimated?.errorMessage;
+    if (!estimated || !estimated?.gasCost) {
+      if (crossChainAction.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE && crossChainAction.useWeb3Provider && crossChainAction.gasCost) {
+        return formatAmountDisplay(
+          `${crossChainAction.gasCost}`,
+          "$"
+        );
```

Comment:
can stay in one line it's fine, single quotes please, no need to wrap as string for `gasCost`

```suggestion
        return formatAmountDisplay(crossChainAction.gasCost, '$');
```

Reply:
right got it